README.md

# NormalityAssessment <img src = "inst/NormalityAssessment/www/NormalityAssessment_Sticker_Image.png" align = "right" height = 190/>

A graphical user interface for testing normality visually

<!-- start badges -->
<!-- start badges -->

![](https://img.shields.io/badge/release-v0.1.0-blue?style=flat) [![CRAN
status](https://www.r-pkg.org/badges/version/NormalityAssessment)](https://CRAN.R-project.org/package=NormalityAssessment)
<!-- end badges -->

## Description

The `NormalityAssessment` `R` package includes an interactive Shiny
application, which is run locally on the user’s machine. It enables the
creation of normal quantile-quantile (QQ) plots and histograms for
assessing normality. The methods implemented are based on recent
developments made in graphical inference. In the app, the features in
the ‘Explore Simulated Data’ tab enable the user to run the Rorschach
procedure, and those in the ‘Include Your Data’ tab allow the user to
run the line-up procedure. Details on these two procedures can be found
in the articles included in the References section below.

## Installation

The `NormalityAssessment` package can be installed from either
<a href="https://cran.r-project.org/" target="_blank">CRAN</a> or
<a href="https://github.com" target="_blank">GitHub</a>.

#### Installing from CRAN

To install from CRAN, run the following code in `R`:

``` r
install.packages("NormalityAssessment")
```

#### Installing from GitHub

To install the package from GitHub, run the following code in `R`:

``` r
install.packages("remotes")  # installs the remotes package for accessing the install_github() function
remotes::install_github("ccasement/NormalityAssessment")  # installs the NormalityAssessment package
```

## Usage

The `NormalityAssessment` application can be run using a single line of
code in `R`:

``` r
NormalityAssessment::runNormalityAssessmentApp()
```

## References

Buja, A., Cook, D., Hofmann, H., Lawrence, M., Lee, E. K., Swayne, D.
F., & Wickham, H. (2009). Statistical inference for exploratory data
analysis and model diagnostics. Philosophical Transactions of the Royal
Society of London A: Mathematical, Physical and Engineering Sciences,
367(1906), 4361-4383.

Majumder, M., Hofmann, H., & Cook, D. (2013). Validation of visual
statistical inference, applied to linear models. Journal of the American
Statistical Association, 108(503), 942-956.

Wickham, H., Cook, D., Hofmann, H., & Buja, A. (2010). Graphical
inference for infovis. IEEE Transactions on Visualization and Computer
Graphics, 16(6), 973-979.

## Bug Reporting

If you happen to find any bugs, we kindly ask that you email us at
<casementc@gmail.com>.

## License

`NormalityAssessment` is distributed under the MIT license. For details,
see the LICENSE.md file.


NEWS.md
# NormalityAssessment 0.1.0

- Added the option to work with built-in data sets of real data in the app's 
  "Include Your Data" tab
- Fixed an issue previously encountered when uploading .mat files
- Updated the information in the app's "About" tab
- Made UI changes to make the app more user-friendly (e.g., changes to the color 
  of certain text)

# NormalityAssessment 0.0.2

- Updated the information in the "About" tab
- Removed old code
- Clarified the license

# NormalityAssessment 0.0.1

- Initial version of the package


R/NormalityAssessment-package.R
#' NormalityAssessment: A Graphical User Interface for Testing Normality
#'   Visually
#'
#' @description The NormalityAssessment package creates plots for assessing
#'   normality. The methods implemented are based on recent development made in
#'   graphical inference. In the app, the features in the 'Explore Simulated
#'   Data' tab enable the user to run the Rorschach procedure, and those in the
#'   'Include Your Data' tab allow the user to run the line-up procedure.
#'
#' @aliases NormalityAssessment-package NormalityAssessment
#'
#' @import shiny
#' @importFrom DT renderDT
#' @importFrom dplyr select_if
#' @importFrom ggplot2 aes element_text facet_wrap geom_histogram ggplot labs
#'   margin stat_qq stat_qq_line theme theme_grey theme_set
#' @importFrom rio import
#' @importFrom rmatio read.mat
#' @importFrom shinyalert shinyalert useShinyalert
#' @importFrom shinyBS bsModal
#' @importFrom stats na.omit pbinom rnorm sd
#' @importFrom stringi stri_dup
#' @importFrom stringr str_trim
#' @importFrom utils head read.csv
#'
#' @section Function: \itemize{
#' \item \code{\link{runNormalityAssessmentApp}}
#' }
#'
#' @details Package: NormalityAssessment \cr
#' Type: Package \cr
#' Version: 0.1.0 \cr
#' Date: 2022-11-04 \cr
#' Depends: R (>= 3.5.0) \cr
#' Imports: dplyr, DT, ggplot2, rio, rmatio, shiny, shinyalert, shinyBS,
#' stringi, stringr
#' License: MIT \cr
#' BugReports: https://github.com/ccasement/NormalityAssessment/issues \cr
#' Encoding: UTF-8 \cr
#'
#' @author
#' Christopher Casement \cr
#' Department of Mathematics \cr
#' Fairfield University \cr
#' \email{casementc@@gmail.com}
#'
#' Laura McSweeney \cr
#' Department of Mathematics \cr
#' Fairfield University
#'
#' @docType package
"_PACKAGE"


R/runNormalityAssessmentApp.R
#' Run the NormalityAssessment Shiny application
#'
#' @description Runs the NormalityAssessment Shiny application.
#'
#' @return There is no return value.
#'
#' @author
#' Christopher Casement \cr
#' Department of Mathematics \cr
#' Fairfield University \cr
#' \email{casementc@@gmail.com}
#'
#' Laura McSweeney \cr
#' Department of Mathematics \cr
#' Fairfield University
#'
#' @references
#' Buja, A., Cook, D., Hofmann, H., Lawrence, M., Lee, E. K., Swayne, D. F.,
#' & Wickham, H. (2009). Statistical inference for exploratory data analysis and
#' model diagnostics. Philosophical Transactions of the Royal Society of London
#' A: Mathematical, Physical and Engineering Sciences, 367(1906), 4361-4383.
#'
#' Majumder, M., Hofmann, H., & Cook, D. (2013). Validation of visual
#' statistical inference, applied to linear models. Journal of the American
#' Statistical Association, 108(503), 942-956.
#'
#' Wickham, H., Cook, D., Hofmann, H., & Buja, A. (2010). Graphical inference
#' for infovis. IEEE Transactions on Visualization and Computer Graphics, 16(6),
#' 973-979.
#'
#' @examples
#' ## only run the app in an interactive R session
#' if (interactive()) {runNormalityAssessmentApp()}
#'
#' @export
runNormalityAssessmentApp <- function() {

  # find and launch app
  run_app_R_script <- '
    appDir <- system.file("NormalityAssessment", package = "NormalityAssessment")

    if (appDir == "") {
      stop("Could not find the NormalityAssessment directory. Try re-installing
        the `NormalityAssessment` package.", call. = FALSE
      )
    }

    shiny::runApp(appDir, launch.browser = TRUE, display.mode = "normal")
  '

  eval(parse(text = run_app_R_script))
}
